diff/parse.go

```diff
@@ -372,8 +372,17 @@ func readline(r *bufio.Reader) (string, error) {
 		return "", err
 	}
 	// consume all remaining line content
-	for isPrefix {
-		_, isPrefix, _ = r.ReadLine()
+	if isPrefix {
+		l := make([]byte, len(line))
+		copy(l, line)
+		for isPrefix {
+			line, isPrefix, err = r.ReadLine()
+			if err != nil {
+				return "", err
+			}
+			l = append(l, line...)
```

Comment:
We don't need this line because the godoc says that `readline` consumes remaining content even if the line size reaches size limit.

> ```go
> // readline reads lines from bufio.Reader with size limit. It consumes
> // remaining content even if the line size reaches size limit.
> func readline(r *bufio.Reader) (string, error) {
> ```

Reply:
So it is intentional to not parse the entire diff if it has long lines?

Reply:
I guess it's yes.

Reply:
Would be great to figure this out, as I am happy to modify the comment if this enhancement is going to be accepted

Reply:
@haya14busa How do you think about this?